### ISSUE TITLE
What SDK do you use

### ISSUE THREAD

**Issue by EricPengShuai** [scivi-tools/scivi.ue.eye#1]
Recently I'm exploring how to get eye position data when watching 360° videos in HTC Vive Pro Eye. I have tried Tobii Pro and OpenVR SDK, but have not obtained complete data.

**Comment by icosaeder** [comment]
We are using SRanipal SDK:
https://developer.vive.com/resources/vive-sense/sdk/vive-eye-and-facial-tracking-sdk/
Which data do you want to retrieve?
This is how we get the eye gaze info:
FFocusInfo focusInfo;
FVector gazeOrigin, gazeTarget;
bool hit = USRanipalEye_FunctionLibrary::Focus(GazeIndex::COMBINE, 1000.0f,
1.0f, m_camera, ECollisionChannel::ECC_WorldStatic, focusInfo, gazeOrigin,
gazeTarget);
This gives us the user's head center position (gazeOrigin) and the gaze hit
point (focusInfo.point). Also, we can check if the user looks at a certain
object (hit && focusInfo.actor == desiredObjectPtr).

пт, 17 сент. 2021 г. в 07:44, Shuai Peng ***@***.***>:

**Comment by EricPengShuai** [comment]
@icosaeder Thank you for your reply.
Could you give me advice on how to run your code to get eye gaze info. Can this code [ReadingTracker/Stimulus.cpp]( https://github.com/scivi-tools/scivi.ue.eye/blob/e9cc04322e835b3940fd9e69b9daa2bb765d883b/ReadingTracker/Stimulus.cpp) get eye gaze info?
Do you try it with HTC Vive Pro Eye device?

**Comment by icosaeder** [comment]
Yes, Stimulus.cpp is the main file where the actor is implemented,
that catches the user's gaze. And yes, we use HTC Vive Pro Eye device as VR
headset.
The idea of what we are doing is described here:
http://sv-journal.org/2021-2/04/
Basically, we have a pretty simple UnrealEngine-based scene, where we place
a kind of a whiteboard with the short text, and the aim is to track the
user's eyes while the user reads that text.
AStimulus class (implemented in Stimulus.cpp) represents this whiteboard.
Do you use Unreal/Unity to manage your virtual scene? If you are using
Unreal Engine as we do, I can provide instructions on how to get HTC Vive
Pro Eye and SRanipal SDK working (the official guide lacks some details
about the project settings you need to tweak to get it working).

пт, 17 сент. 2021 г. в 12:27, Shuai Peng ***@***.***>:
